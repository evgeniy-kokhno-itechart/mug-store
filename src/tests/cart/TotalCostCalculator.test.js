/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TotalCostCalculator from '../../cart/containers/TotalCostCalculator';

configure({ adapter: new Adapter() });

describe('<TotalCostCalculator />', () => {
  let TotalCostCalculatorWrapper;

  beforeEach(() => {
    TotalCostCalculatorWrapper = shallow(<TotalCostCalculator products={[{ currentCurrencyCost: 5 }, { currentCurrencyCost: 10 }]} />);
  });

  it('renders properly with default props', () => {
    expect(TotalCostCalculatorWrapper).toMatchSnapshot();
  });

  it('renders properly with customClasses provided', () => {
    TotalCostCalculatorWrapper.setProps({ customClasses: 'testclass' });
    expect(TotalCostCalculatorWrapper).toMatchSnapshot();
  });

  it('calculates total cost properly', () => {
    expect(TotalCostCalculatorWrapper.first('TotalCost').prop('total')).toEqual(15);
  });
});
