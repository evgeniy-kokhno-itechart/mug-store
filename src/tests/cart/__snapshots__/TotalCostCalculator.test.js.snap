// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalCostCalculator /> renders properly with customClasses provided 1`] = `
<TotalCost
  customClasses="testclass"
  total={15}
/>
`;

exports[`<TotalCostCalculator /> renders properly with default props 1`] = `
<TotalCost
  customClasses=""
  total={15}
/>
`;
